hwilib/commands.py

```diff
@@ -285,10 +289,18 @@ def process_commands(args):
         client = find_device(args)
         if not client:
             return {'error':'Could not find device with specified fingerprint','code':DEVICE_CONN_ERROR}
-    else:
+    elif not args.device_type:
         return {'error':'You must specify a device type or fingerprint for all commands except enumerate','code':NO_DEVICE_PATH}
 
+    try:
```

Comment:
this whole block should only be called if none of the previous above branches are hit. If the user doesn't give the device path they get: `{"error": "'NoneType' object has no attribute 'encode'", "code"`.

Reply:
I've moved it into it's own `elif` block which checks for device type and path.